lib/utils.sh

```diff
@@ -344,6 +344,44 @@ is_kernel_option_enabled() {
     fi
 }
 
+# Detect if module $1 is enabled, can be filtered with $2
+is_kernel_module_enabled() {
+    local MODULE_NAME=$1
+    FNRET=128
+    local module
+
+    if [ $# -eq 2 ]; then
+        module="$($SUDO_CMD modprobe -n -v "$MODULE_NAME" 2>/dev/null | grep -E "$2" | xargs)"
```

Comment:
neither `modprobe` nor `lsmod` work on a monolithic kernel, that's why we prefer to look at /proc/config.gz instead.
This command will fail on a monolithic kernel.
Do you see any benefit doing the way you propose ?

Reply:
If you check CIS debian recommendations (https://www.cisecurity.org/ - CIS_Debian_Family_Linux_Benchmark_v1.0.0) you can find them preferred benchmark method based on kernel module testing and listing but you are right on monolithic kernel this can be an error. I think your actual approach based on file detection in `/lib/modules/$(uname -r)` is problematic because you can have a module overridden (on startup with install or after boot) and be a secure behavior. This can be more secure to check real module dry-run loading too.

Maybe I can merge both approach with 2 options : 

1. remove module detection from `is_kernel_option_enabled` and check `is_kernel_option_enabled` and mine `is_kernel_module_enabled`
2. use only `is_kernel_option_enabled` add `"install /bin/(true|false)"` detection and change detection (modprobe first, file detection fallback)

What do you think ?

Reply:
In fact I think that the second option is the best yes :
* first use your implementation to check with modprobe or lsmod
* in case of failure you'll get something like that
```console
# lsmod
libkmod: ERROR ../libkmod/libkmod-module.c:1657 kmod_module_new_from_loaded: could not open /proc/modules: No such file or directory
Error: could not get list of modules: No such file or directory
```
and you can fallback to the method which check  `/proc/config.gz`

Reply:
do you have on hand the `lsmod` return code on this case ? this is maybe the best way to detect it

Reply:
The return code is 1. The error message is the one I wrote upper.

Reply:
thx

Reply:
implement in 678425dab5d62da84f095c4bbe62abf8646ea163 please note this implementation can impact an inconsistency with sudo flag (example if you don't have right to lsmod as a normal user we will fallback to kernel options detection (mainly false-positive) and with sudo we will check with modprobe that can bring an inconsistency)